Feature/FA-4-Enhance-Project-Setup

ahmedelazab1220 opened the pull request:
<!--📌 Pull Request Template -->

<!--📝 Description -->

This PR enhances the initial project setup by adding essential core utilities, API execution handling, dependency injection (DI), localization updates, and logging configuration. These improvements ensure a more structured and scalable codebase for future development.

**📌 Changes Included**

- 📦 Dependencies: Added required Flutter packages and updated pubspec.yaml.
- 🌍 Localization: Updated Arabic & English translations.
- ⚙ API Execution: Implemented API handling, exception management, and Dio integration.
- 🏗 Dependency Injection: Configured DI with get_it.
- 🛠 Logging & Debugging: Integrated a logger module for better monitoring.
- 🗂 Utilities & Configurations: Added various helper functions and app constants.

## 📌 Pull Request Checklist

### ✅ General
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] The PR title follows the required format: `type/FA-<issue-number>-TaskName`.
- [x] The branch name follows the required format: `feature/meaningful-name`, `fix/issue-description`, etc.

### 📂 Code Quality
- [x] All filenames are in `snake_case`.
- [x] All functions and variable names follow Dart naming conventions.
- [x] All new code is formatted with `dart format .`

### 🛠 CI/CD Validation
- [x] Linting passes with `dart analyze`.
- [x] The branch name is validated against the allowed patterns.
- [x] The PR title is validated against the required format.

### 🚀 Testing
- [x] Unit tests are included for all new features and bug fixes.
- [x] `flutter test --coverage` has been run and the results reviewed.
- [x] The project builds successfully without errors.

### 📝 Additional Information
<!-- Provide any extra context or details about the changes in this PR. -->

### 🔗 Related JIRA Ticket(s):
- [x] JIRA Issue: `FA-4`

### 📸 Screenshots (if applicable):
<!-- Add screenshots to show the changes visually. -->

ahmedelazab1220 committed:
feat(core): add utilities, dependency injection, API execution, and logging setup.

Ahmed-Abdelghany-11 reviewed (approved):
keep going 💪❤️

Ammar720 reviewed (commented):
Set up an API client for handling API requests.

ahmedelazab1220 commented:
> Set up an API client for handling API requests.

ok, i'll add it.
@Ammar720

ahmedelazab1220 committed:
feat(core): Add retrofit client for authentication and update dependencies.

ahmedelazab1220 committed:
feat(core): implement AuthRetrofitClient for API interactions and update dependency configurations and resolve linting.

ahmedelazab1220 committed:
Merge branch 'developement' of github.com:ahmedelazab1220/ecommerce_flower_app into developement

ahmedelazab1220 committed:
fix(linter): ignore unused_field linting rule in analysis_options.yaml

ahmedelazab1220 committed:
feat(routes): add AppRoutes class for route management and initialize routes in main application

ali72-20 commented on lib/core/utils/constants.dart:
```diff
@@ -0,0 +1,5 @@
+class Constants {
```
try make this class with private constructor to avoid any one take an object from it which is no logic

ali72-20 replied:
class Constants{
    Constants._()
}

ali72-20 commented on lib/core/utils/datasource_excution/api_constants.dart:
```diff
@@ -0,0 +1,26 @@
+class ApiConstants {
```
make this class with private constructor to avoid any one to take object from this class

ali72-20 commented on lib/core/utils/dialogs/app_dialogs.dart:
make this class with private constructor

ali72-20 reviewed (approved):
great job keep going 
make this littel comment it's not an issue but to make it better

ahmedelazab1220 committed:
refactor(core): Add private constructors to utility classes
- Ensures consistency by preventing instantiation of:
  - constants.dart
  - api_constants.dart
  - app_dialogs.dart
